### ISSUE TITLE
mpsyt crashes during search (the youtube-dl log stacktrace included)

### ISSUE THREAD

**Issue by viktor-mitin** [ytdl-org/youtube-dl#20556]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2019.04.01*. If it's not, read [this FAQ entry](https://github.com/ytdl-org/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2019.04.01**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/ytdl-org/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/ytdl-org/youtube-dl#faq) and [BUGS](https://github.com/ytdl-org/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

***Note: It looks like the issue 'mpsyt error message. #20310' describes the same problem (not sure).***

- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

Hi, I'm using mpsyt player. Previously it worked well, however some time ago the next issue appeared. When I try to search for 'paganini' text string, mpsyt crashes with the next log:
```
41   Violin Concerto No. 2 in B Minor, Op. 7, MS 48: II. Adagio  07:17
42   Caprice No. 16                        01:42
43   24 Caprices, Op. 1: No. 1 in E Major                                          01:55

Search results for **paganini**                                                                                                                   [1/12>
> ERROR: Signature extraction failed: Traceback (most recent call last):
  File "/home/c/.local/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 1225, in _decrypt_signature  video_id, player_url, s
  File "/home/c/.local/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 1133, in _extract_signature_function  res = self._parse_sig_js(code)
  File "/home/c/.local/lib/python3.6/site-packages/youtube_dl/extractor/youtube.py", line 1200, in _parse_sig_js  initial_function = jsi.extract_function(funcname)
  File "/home/c/.local/lib/python3.6/site-packages/youtube_dl/jsinterp.py", line 245, in extract_function
    raise ExtractorError('Could not find JS function %r' % funcname)
youtube_dl.utils.ExtractorError: Could not find JS function 'encodeURIComponent'; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
 (caused by ExtractorError("Could not find JS function 'encodeURIComponent'; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.",)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'--verbose']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.04.01
[debug] Python version 2.7.15rc1 (CPython) - Linux-4.15.0-46-generic-x86_64-with-Ubuntu-18.04-bionic
[debug] exe versions: ffmpeg 3.4.4, ffprobe 3.4.4
[debug] Proxy map: {}
Usage: youtube-dl [OPTIONS] URL [URL...]

youtube-dl: error: You must provide at least one URL.
Type youtube-dl --help to see a list of all options.
```

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):
- Single video: https://www.youtube.com/watch?v=BaW_jenozKc
- Single video: https://youtu.be/BaW_jenozKc
- Playlist: https://www.youtube.com/playlist?list=PL4lCao7KL_QFVb7Iudeipvc2BCavECqzc

Note that **youtube-dl does not support sites dedicated to [copyright infringement](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free)**. In order for site support request to be accepted all provided example URLs should not violate any copyrights.

---

### Description of your *issue*, suggested solution and other information

Explanation of your *issue* in arbitrary form goes here. Please make sure the [description is worded well enough to be understood](https://github.com/ytdl-org/youtube-dl#is-the-description-of-the-issue-itself-sufficient). Provide as much context and examples as possible.
If work on your *issue* requires account credentials please provide them or explain how one can obtain them.

***Note: It looks like the issue 'mpsyt error message. #20310' describes the same problem (not sure).***

**Comment by viktor-mitin** [comment]
Please describe why did you close the issue:
Why 'incomplete'?
Why did you marked it as 'outdated-version' ?  youtube-dl version in the log is 2019.04.01

Thanks

**Comment by dstftw** [comment]
1. You're using an outdated version from mpsyt.
2. You must provide complete verbose log of failing run not `youtube-dl -v`.

**Comment by viktor-mitin** [comment]
1. I'm using the latest available version of mpsyt '0.2.8'.

c:/srv$ mpsyt --version
mpsyt version      : 0.2.8
   notes           : released 17 February 2018
pafy version       : 0.5.4 (youtube-dl backend)
youtube-dl version : 2018.12.03
Python version     : 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]
Processor          : x86_64
Machine type       : x86_64
Architecture       : 64bit, ELF
Platform           : Linux-4.15.0-46-generic-x86_64-with-Ubuntu-18.04-bionic
sys.stdout.enc     : UTF-8
default enc        : utf-8
Config dir         : /home/c/.config/mps-youtube
env:TERM           : xterm
env:SHELL          : /bin/bash
env:LANG           : en_US.UTF-8
env:LANGUAGE       : en_US

2.  The complete verbose log of failing (including youtube-dl stack trace) is in the issue description.

Thanks

**Comment by dstftw** [comment]
1. Again: you are using an outdated version of youtube-dl from mpsyt.
2. Bother to read issue template of how verbose log should look like.
